File: DataAlgo_forPy/Mock_Challenges/social_network/question2.rb
# Problem
# Given a list of account ID numbers (integers) which contains duplicates , find the one unique integer. (the list is guaranteed to only have one unique (non-duplicated) integer
# Requirements
# Do not use built-in Python functions or methods


def find_unique(array)
  unique_id = 0

  array.each do |num|
    unique_id ^= num
  end

  unique_id
end

p find_unique([5,1,2,1,2,6,7,2,6,7]) 

File: DataAlgo_forPy/Mock_Challenges/social_network/question2.js
// Problem
// Given a list of account ID numbers (integers) which contains duplicates , find the one unique integer. (the list is guaranteed to only have one unique (non-duplicated) integer
// Requirements
// Do not use built-in Python functions or methods


const find_unique = (array) => {
  const seen_numbers = {}

  array.forEach((number) => {
    if (seen_numbers[number]) {
      seen_numbers[number]++
    } else {
      seen_numbers[number] = 1
    }
  })

  return Number(Object.keys(seen_numbers).filter((key) => seen_numbers[key] === 1)[0])
}


const numbers = [5,1,2,1,2,6,7,2,6,7]
console.log(find_unique(numbers)) // 5
